kernel/data-pipeline/dialect/mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/datasource/MySQLJdbcQueryPropertiesExtension.java

```diff
@@ -64,7 +64,13 @@ private static String initMysqlConnectorVersion() {
     }
     
     @Override
-    public Properties extendQueryProperties() {
+    public Properties extendQueryProperties(final Properties props) {
+        if (props.isEmpty()) {
+            return queryProps;
+        }
+        if (props.containsKey("netTimeoutForStreamingResults")) {
+            queryProps.setProperty("netTimeoutForStreamingResults", props.getProperty("netTimeoutForStreamingResults"));
+        }
```

Comment:
Could we add a strategy to decide which one is used, but not hard-coded? e.g. which one is greater for timeout?